connection.go

```diff
@@ -1082,22 +1134,32 @@ func (conn *Connection) contextWatchdog(fut *Future, ctx context.Context) {
 }
 
 func (conn *Connection) send(req Request, streamId uint64) *Future {
+	atomic.AddInt64(&conn.requestCnt, int64(1))
+
 	fut := conn.newFuture(req.Ctx())
 	if fut.ready == nil {
+		atomic.AddInt64(&conn.requestCnt, int64(-1))
```

Comment:
Before the newFuture() we have >1 active requests, after the call we have only 1 request. At now errors inside the call can occur for various reasons.
```suggestion
		if atomic.AddInt64(&conn.requestCnt, int64(-1)) == 0 {
			conn.cond.Broadcast()
		}
```

Reply:
Sorry, I completely forgot about this one. Fixed, logic extracted to a separate function.